aws-databricks/infra.tf

```diff
@@ -71,9 +73,9 @@ resource "aws_route_table" "public" {
 }
 
 resource "aws_route" "public_internet_gateway" {
-  route_table_id         = aws_route_table.public.id
+  route_table_id         = aws_route_table.public[0].id
```

Comment:
Here you need `count` too. Otherwise, if `var.igw_id` is `null` this line will prompt an error as `aws_route_table.public` would be an empty list.